### PR TITLE
feat: TokenImage with no image renders partial token symbol and deterministic dark color

### PR DESCRIPTION
**Pull request by kyhyco**
**What changed? Why?**
- TokenImage with no image renders partial token symbol and deterministic dark color
- fix typo in contribution-guide.mdx

Screenshot 2024-06-06 at 12 20 44 PM

**Notes to reviewers**
Can't use the usual className approach to override `TokenImage` due to adjusting styles depending on prop values

**How has it been tested?**
locally

### CODE REVIEW

File: site/docs/pages/token/token-image.mdx [hunk 1]
```diff
@@ -42,13 +45,29 @@ The `TokenImage` component is an image that crops token image to a circle with a
...
+        blockchain: 'eth',
+        chainId: 8453,
+      }}
+      size={24}
     />
 
     <TokenImage
-      src="https://dynamic-assets.coinbase.com/dbb4b4983bde81309ddab83eb598358eb44375b930b94687ebe38bc22e52c3b2125258ffb8477a5ef22e33d6bd72e32a506c391caa13af64c00e46613c3e5806/asset_icons/4113b082d21cc5fab17fc8f2d19fb996165bcce635e6900f7fc2d57c4ef33ae9.png"
-      size={24}
+      token={{
```
Comment (Zizzamia) [inline site/docs/pages/token/token-image.mdx:61]: Oh brilliant, Token type is the king.

File: site/docs/pages/token/token-image.mdx [hunk 2]
```diff
@@ -60,29 +79,53 @@ The `TokenImage` component is an image that crops token image to a circle with a
...
+    decimals: 6,
+    image: null,
+    blockchain: 'eth',
+    chainId: 8453,
+  }}
+  size={32}
 />
 ```
 
 ```html [return html]
```
Comment (Zizzamia) [inline site/docs/pages/token/token-image.mdx:108]: Thank you for updating those, really appreciate the care for details. 💙

File: src/token/components/TokenImage.tsx [hunk 3]
```diff
@@ -1,7 +1,10 @@
 import { useMemo } from 'react';
 import { TokenImageReact } from '../types';
+import { getTokenImageColor } from './getTokenImageColor';
+
+export function TokenImage({ token, size = 24 }: TokenImageReact) {
+  const { image, symbol, name } = token;
 
-export function TokenImage({ src, size = 24 }: TokenImageReact) {
   const styles = useMemo(() => {
     return {
```
Comment (Zizzamia) [inline src/token/components/TokenImage.tsx:9]: Can all of this be converted in Tailwind vibes? what do you think? or is it not the right approach.

File: src/token/components/getTokenImageColor.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export function getTokenImageColor(str: string) {
```
Comment (Zizzamia) [inline src/token/components/getTokenImageColor.ts:1]: Let's add a test file for this as well.

File: src/token/components/TokenImage.tsx (hunk 3 above)
Comment (kyhyco) [reply src/token/components/TokenImage.tsx:9]: hmmm can do

Comment (kyhyco) [reply src/token/components/TokenImage.tsx:9]: Actually let's do that in a separate PR and discuss it separately.

To not clash with user's tailwind config, we might need to add a prefix to all the tailwind properties which can get a bit messy.

Comment (Zizzamia) [reply src/token/components/TokenImage.tsx:9]: perfect

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**